sprint_4

DenisBurenkov opened the pull request:
Создана верстка и основной функционал приложения основываясь на mock данных

DenisBurenkov committed:
Создана верстка и основной функционал приложения основываясь на Dock данных

andrew-kuzn-yndx commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,208 @@
+    @IBOutlet weak private var imageView: UIImageView!
+    // Обложка фильма
+    @IBOutlet weak private var textLabel: UILabel!
+    // лейбел с вопросом
```
По моему, очень много комментариев к коду. 
Я понимаю, что такие комментарии помогают тебе понимать, что ты пишешь, но после разработки фичи их можно убирать. В репозиторий должен попадать только чистый код)
 Конечно, это не касается критически важных комментариев

andrew-kuzn-yndx commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,208 @@
+        let currentQuestion = questions[currentQuestionIndex]
+        let givenAnswer = false
+        
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
```
На платформе дал совет по этой части кода

andrew-kuzn-yndx commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,208 @@
+            message: result.text,
+            preferredStyle: .alert)
+        
+        let action = UIAlertAction(title: result.buttonText, style: .default) { _ in
```
Тут необходимо прописывать [weak self], чтобы избежать цикла сильных ссылок. 
Можно взять за правило - что всегда, при обращении к self. из замыкания, нужно писать [weak self], чтобы код был безопасным)

DenisBurenkov committed:
1. Убрано излишнее количество коментариев
2. Структуры вынесены из MovieQuizViewController
3. Действие кнопка да и нет вынесенные в отдельный метод
4. Созданный оутлеты для кнопок да и нет
5. Создан метод блокировки кнопок для избежания прокликивания
6. Создан [weak self] в action для избежания цикла сильных ссылок
